docs/design/libraries/ComInterfaceGenerator/ParameterOwnershipAndLifetimes.md

```diff
@@ -0,0 +1,155 @@
+# Ownership and Lifetimes of parameters
+
+## Motivation
+The ComInterfaceGenerator generates both CCW and RCW, and the generated methods on each could fail at many points in the generated stubs. It is important to make sure these failures do not leak memory or lead to double frees.
+This document outlines the lifetimes and ownerships of the parameters as they are passed from managed stubs to COM and vice versa.
+
+## Characteristics
+
+The transfer of ownership depends on the following characteristics of the parameter types
+
+Indirection level
+- ByValue: A value that is not indirected (e.g. `int`, `float`, blittable types in C#)
```

Comment:
```suggestion
- ByValue: A value that is not indirected (for example, `int`, `float`, blittable types in C#)
```

The Microsoft style guides advise against the ["e.g."](https://learn.microsoft.com/style-guide/a-z-word-list-term-collections/e/eg) and ["i.e." ](https://learn.microsoft.com/style-guide/a-z-word-list-term-collections/i/ie)abbreviations.

This should be done throughout the doc.